docs/sql-ref-syntax-qry-select-groupby.md

```diff
@@ -42,42 +42,55 @@ aggregate_name ( [ DISTINCT ] expression [ , ... ] ) [ FILTER ( WHERE boolean_ex
+    `GROUP BY GROUPING SETS ((warehouse), (product))` is semantically equivalent
     to union of results of `GROUP BY warehouse` and `GROUP BY product`. This clause
     is a shorthand for a `UNION ALL` where each leg of the `UNION ALL`
     operator performs aggregation of subset of the columns specified in the `GROUPING SETS` clause.
-
-* **grouping_set**
-
-    A grouping set is specified by zero or more comma-separated expressions in parentheses.
-
-    **Syntax:** `( [ expression [ , ... ] ] )`
-
-* **grouping_expression**
-
-    Specifies the criteria based on which the rows are grouped together. The grouping of rows is performed based on
-    result values of the grouping expressions. A grouping expression may be a column alias, a column position
-    or an expression.
+    Similarly, `GROUP BY GROUPING SETS ((warehouse, product), (product), ())` is semantically
+    equivalent to the union of results of `GROUP BY warehouse, product`, `GROUP BY warehouse`
+    and global aggregate.
+    
+    GROUPING SETS can be followed with normal expressions as well, such as
+    `GROUP BY GROUPING SETS (warehouse, product)`. It generates one grouping set
+    per non-empty subset of the given expressions, so `GROUP BY GROUPING SETS (warehouse, product)`
+    is equivalent to `GROUP BY GROUPING SETS ((warehouse), (product))`.
 
 * **ROLLUP**
 
     Specifies multiple levels of aggregations in a single statement. This clause is used to compute aggregations
     based on multiple grouping sets. `ROLLUP` is a shorthand for `GROUPING SETS`. For example,
-    `GROUP BY warehouse, product WITH ROLLUP` is equivalent to `GROUP BY GROUPING SETS
-    ((warehouse, product), (warehouse), ())`.
+    `GROUP BY warehouse, product WITH ROLLUP` or `GROUP BY ROLLUP(warehouse, product)` is equivalent to
+    `GROUP BY GROUPING SETS((warehouse, product), (warehouse), ())`.
+    `GROUP BY ROLLUP(warehouse, product, (warehouse, location))` is equivalent to
+    `GROUP BY GROUPING SETS((warehouse, product, location), (warehouse, product), (warehouse), ())`.
     The N elements of a `ROLLUP` specification results in N+1 `GROUPING SETS`.
 
 * **CUBE**
 
     `CUBE` clause is used to perform aggregations based on combination of grouping columns specified in the
     `GROUP BY` clause. `CUBE` is a shorthand for `GROUPING SETS`. For example,
-    `GROUP BY warehouse, product WITH CUBE` is equivalent to `GROUP BY GROUPING SETS
-    ((warehouse, product), (warehouse), (product), ())`.
+    `GROUP BY warehouse, product WITH CUBE` or `GROUP BY CUBE(warehouse, product)` is equivalent to 
+    `GROUP BY GROUPING SETS((warehouse, product), (warehouse), (product), ())`.
+    `GROUP BY CUBE(warehouse, product, (warehouse, location))` is equivalent to
+    `GROUP BY GROUPING SETS((warehouse, product, location), (warehouse, product), (warehouse, location),
+     (product, warehouse, location), (warehouse), (product), (warehouse, product), ())`.
```

Comment:
`(warehouse, product)` is duplicated, is it OK?

Reply:
> `(warehouse, product)` is duplicated, is it OK?

It's right

Reply:
> `(warehouse, product)` is duplicated, is it OK?

Each grouping_set will be a subset of result
![image](https://user-images.githubusercontent.com/46485123/112616090-4758bd00-8e5e-11eb-933f-cab866d7bdcc.png)